src/dataset/impl_dataset.rs

```diff
@@ -1091,6 +1091,32 @@ where
     }
 }
 
+impl<F: Float, D, S, O> PredictRef<ArrayBase<D, Ix2>, Array1<S>> for O
+where
+    D: Data<Elem = F>,
+    O: PredictInto<ArrayBase<D, Ix2>, Array1<S>>,
+    S: Default,
+{
+    fn predict_ref(&self, records: &ArrayBase<D, Ix2>) -> Array1<S> {
+        let mut targets = Array1::default(records.nrows());
+        self.predict_into(records, &mut targets);
+        targets
+    }
+}
+
+impl<F: Float, D, S, O> PredictRef<ArrayBase<D, Ix2>, Array2<S>> for O
+where
+    D: Data<Elem = F>,
+    O: PredictInto<ArrayBase<D, Ix2>, Array2<S>>,
+    S: Default,
+{
+    fn predict_ref(&self, records: &ArrayBase<D, Ix2>) -> Array2<S> {
+        let mut targets = Array2::default((records.nrows(), 1));
```

Comment:
Could there be models that predict into an `Array2` that has a shape other than `(records.nrows(), 1)`?
If there are such cases, I am not sure how we could deal with them. Any ideas?

Reply:
yes for example partial least squares have multiple targets, add a function `num_targets` to your trait set?